apps/glusterfs/operations_block_volume.go

```diff
@@ -486,7 +488,10 @@ func (bve *BlockVolumeExpandOperation) Build() error {
 		}
 
 		requiredFreeSize := bve.newSize - bv.Info.Size
-		if bhv.Info.BlockInfo.FreeSize < requiredFreeSize {
+		if requiredFreeSize == 0 {
+			logger.Info("Attempting to fix failures seen by recent expand operation")
```

Comment:
This log line would confuse me if I was debbuging things and saw it w/o looking at the code. I would 1. make this a warning to highlight that the state of the system is a tad bit out of whack and 2. reword it to clarify what is actually being done. Perhaps, "Re-executing block volume expansion on [%v]: volume not fully expanded on backend" where %v is to be replaced by the volume id.

Reply:
ok

Reply:
Oh there was a thread about this log message already.

I replied as a separate comment that this log line might give an illusion to the user that there is some kind of retry mechanism. My suggestion would be something like:

"executing block volume expand on [%v] as usable size is not same as the size."